src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs

```diff
@@ -7047,6 +7047,158 @@ public void Create_Populates_EndpointBuilderWithRequestDelegateAndMetadata()
         Assert.Same(options.EndpointBuilder.Metadata, result.EndpointMetadata);
     }
 
+    private class ParameterListRequiredStringFromDifferentSources
+    {
+        public HttpContext? HttpContext { get; set; }
+
+        [FromRoute]
+        public required string RequiredRouteParam { get; set; }
+
+        [FromQuery]
+        public required string RequiredQueryParam { get; set; }
+
+        [FromHeader]
+        public required string RequiredHeaderParam { get; set; }
+    }
+
+    [Fact]
+    public async Task RequestDelegateFactory_AsParameters_SupportsRequiredMember()
+    {
+        // Arrange
+        static void TestAction([AsParameters] ParameterListRequiredStringFromDifferentSources args) { }
+
+        var httpContext = CreateHttpContext();
+
+        var factoryResult = RequestDelegateFactory.Create(TestAction);
+        var requestDelegate = factoryResult.RequestDelegate;
+
+        // Act
+        await requestDelegate(httpContext);
+
+        // Assert that the required modifier disables optionality
```

Comment:
If I am not mistaken, this test works not because the `required` attribute and instead because the `NullabilityState` is `NotNull` right? If so, maybe we should update this comment to make it clearer.